internal/osbuild/skopeo_stage.go

```diff
@@ -37,3 +39,20 @@ func NewSkopeoStage(path string, images ContainersInput, manifests *FilesInput)
 		Inputs: inputs,
 	}
 }
+
+// NewSkopeoStageSourceFilesInputs creates a FilesInput that can be used with
+// the NewSkopeoStage to download manifest lists. If there are no list digests
+// in the container specs, it returns nil.
+func NewSkopeoStageSourceFilesInputs(containers []container.Spec) *FilesInput {
```

Comment:
Not sure about this.  Technically, it just produces an `org.osbuild.files` input and that could be added to any stage that can take that type of input.  The mistake here is defining this function in `skopeo_stage.go` instead of somewhere else, perhaps `osbuild/files_input.go`.  I think it would make sense to move it there and keep `NewContainersInputForSources` in `osbuild/containers_input.go`, making both constructors for a certain type of input defined in the file for that input.

Reply:
Moved to `files_input.go` and renamed to `NewFilesInputForManifestLists()`.

Reply:
I checked what we do for other staged and we seem to produce the actual stage-specific inputs -> https://github.com/osbuild/osbuild-composer/blob/da2e80f99f65bf708e0c7c3a31ee4844846287d6/internal/osbuild/rpm_stage.go#L123-L126

So maybe we could have something like `NewSkopeoStageInputs()` which will take in `[]container.Spec` as an argument and it will produce `SkopeoStageInputs` instead. It probably makes sense to always produce the `Images` and `ManifestLists` inside it from the same slice of `container.Spec` 🤔

Reply:
Anyway, we are not consistent, so your approach is fine by me.